js/plugins/iq.js

```diff
@@ -236,7 +246,29 @@ var IqPlugin= function (framework)
         this.ooyalaReporter.reportReplay();
         OO.log("IQ: Reported: reportReplay()");
         break;
-
+      //OO.EVENTS.BUFFERING -> OO.Analytics.EVENTS.VIDEO_BUFFERING_STARTED.
+      case OO.Analytics.EVENTS.VIDEO_BUFFERING_STARTED: 
+        eventParams = params[0];
+        eventMetadata = {};
+        eventMetadata.qosEventName = eventName;
+        eventMetadata.position = eventParams.position;
+        OO.log("IQ: Reported: reportCustomEvent() for event: " + eventName + " with args:" + JSON.stringify(eventMetadata));
+        this.ooyalaReporter.reportCustomEvent(eventName, eventMetadata);
+        break;
+      case OO.Analytics.EVENTS.INITIAL_PLAY_STARTING:
+      case OO.Analytics.EVENTS.PLAYBACK_READY:
+      case OO.Analytics.EVENTS.API_ERROR:
+      case OO.Analytics.EVENTS.BITRATE_INITIAL:
+      case OO.Analytics.EVENTS.BITRATE_FIVE_SEC:
+      case OO.Analytics.EVENTS.BITRATE_STABLE:
+      case OO.Analytics.EVENTS.PLAYBACK_START_ERROR:
+      case OO.Analytics.EVENTS.PLAYBACK_MIDSTREAM_ERROR:
+      case OO.Analytics.EVENTS.PLUGIN_LOADED:
+        eventMetadata = params[0];
+        eventMetadata.qosEventName = eventName;
+        OO.log("IQ: Reported: reportCustomEvent() for event: " + eventName + " with args:" + JSON.stringify(eventMetadata));
+        this.ooyalaReporter.reportCustomEvent(eventName, eventMetadata);
+        break;
```

Comment:
Can we add unit tests for these new event handlers?

Reply:
Do you mean something to test reportCustomEvent(), or do you want a unit test for each event checked by the switch case statement?

Reply:
Each event checked by the switch statement.

Reply:
@aeng7 added a new test for each of the qos events, working on ad events now

Reply:
Awesome, thanks!

Reply:
@aeng7 Added a bunch of unit tests that were missing for the ad events, should be good for unit tests now